robots/cmd/kubevirtci-presubmit-remover/main.go

```diff
@@ -0,0 +1,171 @@
+	"golang.org/x/oauth2"
+	"io/ioutil"
+	"k8s.io/test-infra/prow/config"
+	"kubevirt.io/project-infra/robots/pkg/kubevirt/release"
+	"os"
+	"sigs.k8s.io/yaml"
+
+	"kubevirt.io/project-infra/robots/pkg/querier"
+)
+
+const OrgAndRepoForJobConfig = "kubevirt/kubevirtci"
+
+type options struct {
+	port int
+
+	dryRun bool
+
+	TokenPath                        string
+	endpoint                         string
+	jobConfigPathKubevirtciPresubmit string
+}
+
+func (o *options) Validate() error {
+	if _, err := os.Stat(o.jobConfigPathKubevirtciPresubmit); os.IsNotExist(err) {
+		return fmt.Errorf("jobConfigPathKubevirtciPresubmit is required: %v", err)
+	}
+	return nil
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.BoolVar(&o.dryRun, "dry-run", true, "Whether the file should get modified or just modifications printed to stdout.")
+	fs.StringVar(&o.TokenPath, "github-token-path", "/etc/github/oauth", "Path to the file containing the GitHub OAuth secret.")
+	fs.StringVar(&o.endpoint, "github-endpoint", "https://api.github.com/", "GitHub's API endpoint (may differ for enterprise).")
+	fs.StringVar(&o.jobConfigPathKubevirtciPresubmit, "job-config-path-kubevirtci-presubmit", "", "The directory of the k8s providers")
+	err := fs.Parse(os.Args[1:])
+	if err != nil {
+		fmt.Println(fmt.Errorf("failed to parse args: %v", err))
+		os.Exit(1)
+	}
+	return o
+}
+
+func main() {
+
+	logrus.SetFormatter(&logrus.JSONFormatter{})
+	// TODO: Use global option from the prow config.
+	logrus.SetLevel(logrus.DebugLevel)
+	log := logrus.StandardLogger().WithField("robot", "kubevirtci-presubmit-remover")
+
+	o := gatherOptions()
+	if err := o.Validate(); err != nil {
+		log.WithError(err).Error("Invalid arguments provided.")
+		os.Exit(1)
+	}
+
+	ctx := context.Background()
+	var client *github.Client
+	if o.TokenPath == "" {
+		var err error
+		client, err = github.NewEnterpriseClient(o.endpoint, o.endpoint, nil)
+		if err != nil {
+			log.Panicln(err)
+		}
+	} else {
+		token, err := ioutil.ReadFile(o.TokenPath)
+		if err != nil {
+			log.Panicln(err)
+		}
+		ts := oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: string(token)},
+		)
+		client, err = github.NewEnterpriseClient(o.endpoint, o.endpoint, oauth2.NewClient(ctx, ts))
+		if err != nil {
+			log.Panicln(err)
+		}
+	}
+
+	jobConfig, err := config.ReadJobConfig(o.jobConfigPathKubevirtciPresubmit)
+	if err != nil {
+		log.Panicln(err)
+	}
+
+	releases, _, err := client.Repositories.ListReleases(ctx, "kubernetes", "kubernetes", nil)
+	if err != nil {
+		log.Panicln(err)
+	}
+	releases = querier.ValidReleases(releases)
+	if len(releases) == 0 {
+		log.Info("No release found, nothing to do.")
+		os.Exit(0)
+	}
+
+	latestMinorReleases := release.GetLatestMinorReleases(release.AsSemVers(releases))
+	if len(latestMinorReleases) < 3 {
+		log.Info("Not enough minor releases found, nothing to do.")
+	}
+
+	targetRelease := latestMinorReleases[3:4][0]
```

Comment:
@dhiller Hi. I can't differ between access of `latestMinorReleases[3:4][0]` and `latestMinorReleases[3]`, will it return a different element? is there some implicit sorting that happens when you create the temporary list with `latestMinorReleases[3:4]` and then take the first element?

Reply:
TBH I have no idea why I did this in the most complicated way :-D

Reply:
Fixed. Thank you for catching that!